pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -363,6 +375,26 @@ func updateNetwork(vmi *k6tv1.VirtualMachineInstance, vmStats *stats.DomainStats
 	}
 }
 
+func updateTelemetry(nodeName string, vmis []*k6tv1.VirtualMachineInstance, ch chan<- prometheus.Metric) {
+	phasesMap := make(map[string]uint64)
+
+	for _, vmi := range vmis {
+		phasesMap[strings.ToLower(string(vmi.Status.Phase))] += 1
+	}
+
+	for phase, count := range phasesMap {
+		mv, err := prometheus.NewConstMetric(
+			vmiCountDesc, prometheus.GaugeValue,
+			float64(count),
+			nodeName, phase,
+		)
+		if err != nil {
+			continue
```

Comment:
Can add some logging here ?

Reply:
We can, but is a tough call. The `prometheus.NewConstMetric` is supposed to never fail, so failure is exceptional and would deserve a log. But if this happen repeteadly, we can spam the logs, and this is something we want to avoid.

Reply:
So maybe debug level ?

Reply:
Uhm, good point. This maybe another case on which better to change all the occurrence in one go. If so, I'll file a new PR and link it here. Otherwise I'll add another commit.

Reply:
I decided to tackle all the cases at once: https://github.com/kubevirt/kubevirt/pull/2523